identity-apps-core/apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources_pt_PT.properties

```diff
@@ -215,7 +215,7 @@ your.password.should.be.between=Sua senha deve estar entre
 must.be.alphanumeric=Deve ser uma string alfanumérica (a-z, A-Z, 0-9) entre
 to=para
 characters.including.one.letter=caracteres incluindo pelo menos uma letra.
-characters.may.contain=caracteres de comprimento, incluindo pelo menos uma letra, e pode conter uma combinação dos seguintes caracteres: a-z, A-Z, 0-9, !@#$%&'*+\\\\=?^_.{|}~ -.
+characters.may.contain=caracteres de comprimento, incluindo pelo menos uma letra, e pode conter uma combinação dos seguintes caracteres: a-z, A-Z, 0-9, !@#$&'+\\\\=^_.{|}~ -.
```

Comment:
It will introduce more escape characters to this sequence and might look ugly IMO.

Reply:
> It will introduce more escape characters to this sequence and might look ugly IMO.

All you need to do is use a placeholder such as {{specialCharListStr}} in the places where special char list needs to be added.

eg: 
```
characters.may.contain=caracteres de comprimento, incluindo pelo menos uma letra, e pode conter uma combinação dos seguintes caracteres: {{specialCharListStr}}.
```

And in the code, you just replace the placeholder with the char list, as follows.

```java
String ALLOWED_SPECIAL_CHARS = "a-z, A-Z, 0-9, !@#$&'+\\\\=^_.{|}~ -";
IdentityManagementEndpointUtil.i18n(recoveryResourceBundle, "characters.may.contain").replace("{{specialCharListStr}}",ALLOWED_SPECIAL_CHARS)
```

I couldn't understand how that approach can make the implementation ugly :)